Be able to set options via YYLEX_PARAM, useful when using reentrant flex.

--- byacc-20100216.orig/skeleton.c
+++ byacc-20100216/skeleton.c
@@ -55,6 +55,17 @@ const char *xdecls[] =
     "",
     "extern int YYPARSE_DECL();",
     "",
+    "/* compatibility with flex */",
+    "#ifdef YYLEX_PARAM",
+    "#define YYLEX_DECL() yylex(void *YYLEX_PARAM)",
+    "#define YYLEX yylex(YYLEX_PARAM)",
+    "#else",
+    "#define YYLEX_DECL() yylex(void)",
+    "#define YYLEX yylex()",
+    "#endif",
+    "",
+    "extern int YYLEX_DECL();",
+    "",
     0
 };
 
@@ -232,7 +243,7 @@ const char *body_2[] =
     "    if ((yyn = yydefred[yystate]) != 0) goto yyreduce;",
     "    if (yychar < 0)",
     "    {",
-    "        if ((yychar = yylex()) < 0) yychar = 0;",
+    "        if ((yychar = YYLEX) < 0) yychar = 0;",
     "#if YYDEBUG",
     "        if (yydebug)",
     "        {",
@@ -368,7 +379,7 @@ const char *trailer[] =
     "        *++yystack.l_mark = yyval;",
     "        if (yychar < 0)",
     "        {",
-    "            if ((yychar = yylex()) < 0) yychar = 0;",
+    "            if ((yychar = YYLEX) < 0) yychar = 0;",
     "#if YYDEBUG",
     "            if (yydebug)",
     "            {",
